src/app/hm-admin/services/schema/schema.service.ts

```diff
@@ -0,0 +1,171 @@
+import 'rxjs/add/operator/last';
+import 'rxjs/add/operator/zip';
+import * as _ from 'lodash';
+import {ConfigService} from '../config/Config.service';
+import {entrypointHelper} from '../../helper/entrypointHelper';
+import {schemaHelper} from '../../helper/schemaHelper';
+import {EntryPoint} from '../models/EntryPoint';
+import {stringHelper} from '../../helper/stringHelper';
+import { CoreDefinitionService } from '../core/CoreDefinition.service';
+
+@Injectable()
+export class SchemaService {
+
+  schema: BehaviorSubject<any>;
+  entrypoints: BehaviorSubject<Array<EntryPoint>>;
+  availableEntrypoints: Array<EntryPoint> = [];
+
+  constructor(private http: Http,
+              private config: ConfigService,
+              private coreDefinitionService: CoreDefinitionService) {
+    this.schema = new BehaviorSubject(null);
+    this.entrypoints = new BehaviorSubject([]);
+
+  }
+
+  /**
+   * load API description
+   */
+  load() {
+    this.coreDefinitionService.getDefinitions().subscribe((definition) => {
+      this.loadRoot();
+      this.loadDocumentation(definition);
+    });
+
+  }
+
+  /**
+   * load available entrypoints from API's root
+   * @private
+   */
+  loadRoot() {
+    let request: string = this.config.api.baseUrl;
+    let entrypoints = this.http.get(request)
+      .map(data => data.json())
+      .map(entrypointHelper.filterEntryPoints);
+
+    entrypoints.subscribe(
+      (entrypoints => {
+        _.forEach(_.keys(entrypoints), (key) => {
+          let cleanModel = stringHelper.toCamelCase(entrypointHelper.getModelById(key)),
+            entrypoint: EntryPoint = entrypointHelper.getEntryPointByModel(key, this.availableEntrypoints);
+
+          if (undefined === entrypoint) {
+            entrypoint = new EntryPoint(cleanModel);
+            this.availableEntrypoints.push(entrypoint);
+          }
+
+          entrypoint.url = entrypoints[key];
+        });
+        this.entrypoints.next(this.availableEntrypoints);
+      }),
+      () => {
+        throw new Error(`API's url is unreachable: ${request}.`);
+      }
+    );
+  }
+
+  /**
+   * load documentation about the API
+   */
+  loadDocumentation(definition: any) {
+    let request = [this.config.api.baseUrl, 'vocab'].join('/'),
+      apidoc = this.http.get(request)
+        .map(data => data.json())
+        .share();
+
+    apidoc.subscribe(
+      null,
+      () => {
+        throw new Error(`API's url is unreachable: ${request}.`);
+      }
+    );
+
+    let api = apidoc.map(data => data[definition.supportedClass])
+      .map(entrypointHelper.filterSupportedClass)
+      .switchMap(results => results);
+
+    this.loadCollectionOperations(api, definition);
+    this.loadSingleOperations(api, definition);
+    this.loadSchema(apidoc, definition);
+
+  }
+
+  /**
+   * Load collection operations
+   *
+   * @param {Observable<any>} api
+   * @param {string} request
+   * @private
+   */
+  loadCollectionOperations(api: Observable<any>, definition: any) {
```

Comment:
We should define an interface for the objects carried by the api’s parameter observable since it seems we know them well (we are using the `@id` property of such object with the following lines.).